File: solutions/2871. Split Array Into Maximum Number of Subarrays/2871.py
class Solution:
  def maxSubarrays(self, nums: list[int]) -> int:
    ans = 0
    score = 0

    for num in nums:
      score = num if score == 0 else score & num
      if score == 0:
        ans += 1

    return max(1, ans)

File: solutions/2871. Split Array Into Maximum Number of Subarrays/2871.java
class Solution {
  public int maxSubarrays(int[] nums) {
    int ans = 0;
    int score = 0;

    for (final int num : nums) {
      score = score == 0 ? num : score & num;
      if (score == 0)
        ++ans;
    }

    return Math.max(1, ans);
  }
}
